handler/proxy_client.go

```diff
@@ -58,6 +59,10 @@ func (p *ProxyClient) sign(req *http.Request, service *endpoints.ResolvedEndpoin
 }
 
 func (p *ProxyClient) Do(req *http.Request) (*http.Response, error) {
+	overwriteHost := os.Getenv("SERVICE_HOST")
+	if overwriteHost != "" {
+		req.Host = overwriteHost
+	}
```

Comment:
It seems like this is something you might want to pass in to the ProxyClient at instantiation time - rather than reading it from your env on every request.  Also - the proxy reads the host from the `Host` HTTP header - so you should be able to achieve this desired behavior by overwriting the `Host` header on the incoming request.  Check out `proxy_url`in Promtheus https://prometheus.io/docs/prometheus/latest/configuration/configuration/

Reply:
It seems like this is something you might want to pass in to the ProxyClient at instantiation time - rather than reading it from your env on every request
-- thanks for the advice, i will later update to setup the env provider save the env in memory at instantiation time

so you should be able to achieve this desired behavior by overwriting the `Host` header on the incoming request.
-- i do think about setup a reverse proxy for overwriting the host header, but I also think setting up the `upstream` is kind of normal for proxy, and overwriting the host header here is kind of similar way to set the upstream,, 
and also save the overhead of the additional reverse proxy

Reply:
@tantona I moved reading ENV to ProxyClient  instantiation time. 
I am newbie to golang, feel free to comment my changes